### DISCUSSION TITLE
🐛 BUG: custom-attributes editor prompt not working

### DISCUSSION THREAD

**Discussion by J-env** [withastro/language-tools#843] [Feature requests]
### Describe the Bug

Set custom properties according to the official documentation, but the editor prompt does not work. [astro docs](https://docs.astro.build/en/guides/typescript/#built-in-html-attributes)

<img width="821" alt="image" src="https://github.com/withastro/language-tools/assets/25259920/a1db281b-677d-4b0b-bd7d-5e5f746ce8f6">

### Steps to Reproduce

1. `npm init astro` using template <NAME>
2. create file `src/custom-attributes.d.ts` 
3. Enter data- attribute in astro component, editor prompt does not work

**Comment by Princesseuh** [comment]
Completions for HTML attributes is powered by [custom data](https://code.visualstudio.com/blogs/2020/02/24/custom-data-format) instead of types. It's done that way because HTML custom data has generally much richer information available in both how to complete the completion, the attributes value or the hover description.

It'd be nice to support somehow both types and custom data, but it's quite challenging notably because of the speed difference in getting the HTML completions vs the TS ones, not to mention the duplicate completions they'd provide.

Since this is not a bug, and more so a feature request, I'll move this to the discussion area, thank you!
